hydra_agent/redis_core/graphutils_operations.py

```diff
@@ -5,32 +5,36 @@
 from hydra_agent.redis_core.redis_proxy import RedisProxy
 from hydra_agent.redis_core.graphutils import GraphUtils
 from redisgraph import Graph, Node
+from requests import Session
 
 logger = logging.getLogger(__file__)
 
 
 class GraphOperations():
 
-    def __init__(self, entrypoint_url: str, redis_proxy: RedisProxy):
+    def __init__(self, entrypoint_url: str, api_doc: dict,
+                 redis_proxy: RedisProxy):
         self.entrypoint_url = entrypoint_url
+        self.api_doc = api_doc
         self.redis_proxy = redis_proxy
         self.redis_connection = redis_proxy.get_connection()
         self.vocabulary = 'vocab'
         self.graph_utils = GraphUtils(redis_proxy)
         self.redis_graph = Graph("apigraph", self.redis_connection)
+        self.session = Session()
```

Comment:
`Session` shouldn't be called here. For any call that needs a session the code should call a function in the agent class. So probably you wan to create a method in the other class, and make this class to call it. In the current way, segregation of concerns is violated.

Reply:
You are right, this would get messed up in the future. But I cannot simply instantiate a new Agent and call it, because it will be a new session and not inherit the Agent session's attributes.

What's the right way to call the Agent from here persisting the session since this module is instantiated by the Agent?

Reply:
Session are really volatile. you can have a classmethod in Agent like `Agent.get_session()`, that just instantiate and return a session, without assigning to `self.session`. I meant, use an helper, do not assign a session to the operations instance, otherwise every instance will need to carry around a session without needing it. Factorise the functionality, so to avoid to add the functionality to all the instances.

Reply:
Okay, I also researched some design patterns and got some ideas. I'll just finish up a PR about updating Redis graph for #112 and then I'll propose a solution for this.